tango/server/report.js

```diff
@@ -0,0 +1,299 @@
+            <td align="left"><font color="${color}">${text}</font></td>
+            <td></td>
+          </tr>
+          `;
+};
+
+// return a basic date
+const formatDate = (label, date) => {
+  if (!date) {
+    return "";
+  }
+  return `
+        <tr>
+            <td></td>
+            <td align="left">${dayjs(date).format("YYYY-MM-DD h:mm:ss A")}</td>
+            <td></td>
+          </tr>
+          `;
+};
+
+// return a date range
+const formatDateRange = (startDate, endDate) => {
+  if (!startDate) {
+    return "";
+  }
+
+  const dayFormat = "YYYY-MM-DD";
+  const timeFormat = "h:mm:ss A";
+  const fullFormat = `${dayFormat} ${timeFormat}`;
+  let endFormat = fullFormat;
+  if (dayjs(startDate).format(dayFormat) === dayjs(endDate).format(dayFormat)) {
+    // if the same day, jsut show the time difference
+    endFormat = timeFormat;
+  }
+
+  return `
+        <tr>
+            <td></td>
+            <td align="left">${dayjs(startDate).format(fullFormat)} - ${dayjs(
+    endDate
+  ).format(endFormat)}</td>
+          <td></td>
+          </tr>
+          `;
+};
+
+// return a blue text with a href
+const formatLink = (label, url) => {
+  if (!url) {
+    return "";
+  }
+  return `
+        <tr>
+          <td></td>
+          <td align="left" href="${url}" target="_blank"><font color="${
+    colors.N9
+  }">${label}:</font>${" "}<font color="${colors.B6}">${url}</font></td>
+          <td></td>
+        </tr>`;
+};
+
+// return a key (has a special id for copy to clipboard later)
+// the `id="key;${value}" href=" "` is a hack to allow selecting ny id later
+const formatKey = (value) => {
+  if (!value) {
+    return "";
+  }
+  return `
+        <tr>
+          <td></td>
+          <td align="left" id="key;${value}" href=" ">${value}</td>
+          <td></td>
+        </tr>`;
+};
+
+// return basic text
+const formatText = (label, value) => {
+  if (!value) {
+    return "";
+  }
+  return `
+        <tr>
+          <td></td>
+          <td align="left">${label}${label ? ": " : ""}${value}</td>
+          <td></td>
+        </tr>`;
+};
+
+// convert the data to a node
+const getTable = (data) => {
+  const isOpen = state.openMap[data.unique_id];
+  let ret = `label=<
+        <table cellspacing="0" cellpadding="2" border="0" color="${
+          colors.B10
+        }" cellborder="0" bgcolor="${colors.white}">
+          <tr>
+            <td align="left" bgcolor="${colors.B10}" id="expando;${
+    data.unique_id
+  }" href=" "><img src="${
+    isOpen ? "/close.svg" : "/open.svg"
+  }" /></td>
+            <td bgcolor="${colors.B10}" ><font point-size="16" color="${
+    colors.N2
+  }">${data.step_name}</font></td>
+        <td bgcolor="${colors.B10}">${"   "}</td>
+          </tr>
+          <!-- Some extra space at the top -->
+          ${isOpen ? `<tr><td>${" "}</td></tr>` : null}
+          ${isOpen ? formatState(data, true) : formatState(data)}
+          ${isOpen ? formatText(undefined, data.unique_id) : null}
```

Comment:
I don't know what changed, but this "undefined" now renders on the page.

Reply:
see above